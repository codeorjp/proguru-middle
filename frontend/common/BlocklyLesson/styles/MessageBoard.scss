@import 'common/styles/variables.scss';
.messageboard {
  width: 100%;
  height: 100%;
  position: relative;
  outline: none;

  .messageboardWindow {
    margin: 0 auto;
    height: 100%;

    .messageboardHeader {
      border-radius: 10px 10px 0 0;
    }

    .messageboardBody {
      background: #fff;
      padding: 20px;
    
      .messageboardBodyHeader {
        margin-bottom: 1px;
      }

      .messageboardFooterContents {
        width: 90%;
        margin: 0 auto;
        padding-top: 15px;
        padding-bottom: 15px;
      }

      .messageboardName {
        text-align: center;
        padding-bottom: 15px;
        font-size: 1.4rem;
      }

      .messageboardContents {
        padding: 10px 10px;
        height: 400px;
        overflow-y: scroll;
        overflow-x: hidden;
      }

      .messageboardMoreButton {
        background: $blue-1;
        border: 0;
        padding: 10px 20px;
        border-radius: 100px;
        color: #fff;
      }
    }
  }
}

.sendMessage {
  text-align: right;
  margin: 10px 0;
}

.receiveMessage {
  text-align: left;
  margin: 10px 0;
}

.overlay {
  border-radius: 1px;
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 100%;
  z-index: 100;
  color: white;
  background: rgba(#000, 0.7);
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;

  p {
    font-size: 22px;
  }
}

.displayNone {
  display: none;
}
