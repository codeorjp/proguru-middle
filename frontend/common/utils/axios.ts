import axios from "axios";
import { csrfToken } from "rails-ujs";

axios.defaults.headers.common["X-CSRF-TOKEN"] = csrfToken();
axios.defaults.headers.common["X-Requested-With"] = "XMLHttpRequest";
axios.defaults.headers.common["Content-Type"] = "application/json";

export default axios;
