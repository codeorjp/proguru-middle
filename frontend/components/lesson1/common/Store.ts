import { action, decorate, observable, toJS } from "mobx";
import Storage from "components/lesson1/common/util/Storage";
import { DesignSchemaType } from "common/types/DesignSchemaType";
import { ComponentStyleType } from "common/types/ComponentStyleType";
import { LessonType } from "common/types/LessonType";
import { StageType } from "common/types/StageType";

type ClickElement = {
  title: string;
  className: string;
  style: ComponentStyleType;
  editDesignComponent: string[];
};

type StageConfig = {
  className: string;
  editDesignComponent: string[];
}[];

class Store {
  storage: Storage;
  schemas: DesignSchemaType;
  clickElm: ClickElement;
  stageConfig: StageConfig;
  sendSpeechText: string;
  receiveSpeechText: string;
  isFirstTime: boolean;
  isStageFirstTime: boolean;

  constructor(
    public userId: number | string,
    public lesson: LessonType,
    public stage: StageType,
    public submittedWorkspace: any
  ) {
    this.storage = new Storage(`${window.location.href}-user${this.userId}`);
    this.schemas = {
      SendSpeechBalloon: {
        style: {
          font: { fontSize: "16px", textAlign: "right" },
          color: "#000",
          border: {},
          background: "",
          borderRadius: "",
        },
      },
      SendProfileIcon: {
        style: {
          imgSize: {
            width: "200px",
            height: "200px",
          },
          border: {
            borderWidth: "1px",
          },
          borderRadius: "",
          background: "#fff",
        },
      },
      ReceiveSpeechBalloon: {
        style: {
          font: { fontSize: "16px", textAlign: "right" },
          color: "#000",
          border: {},
          borderRadius: "",
          background: "",
        },
      },
      ReceiveProfileIcon: {
        style: {
          imgSize: {
            width: "200px",
            height: "200px",
          },
          border: {
            borderWidth: "1px",
          },
          borderRadius: "",
          background: "#fff",
        },
      },
      ChatInput: {
        style: {
          font: { fontSize: "14px" },
          color: "#000",
          border: {},
          borderRadius: "",
          background: "#fff",
          padding: "",
        },
      },
      SubmitButton: {
        style: {
          font: { fontSize: "14px" },
          color: "#000",
          border: {},
          borderRadius: "",
          background: "#fff",
          padding: "",
        },
      },
      ChatBody: {
        style: {
          background: "#fff",
        },
      },
    };
    this.clickElm = {
      title: "", // 編集するコンポーネントタイトル
      className: "", // CSSクラス名
      style: {}, // 当てるスタイル
      editDesignComponent: [], // 編集コンポーネント
    };
    this.stageConfig = [
      {
        // アタッチするデザイン編集項目
        className: "", // CSSクラス名
        editDesignComponent: [], // 編集コンポーネント
      },
    ];
    this.sendSpeechText = "こんにちは，私の名前は美咲です。";
    this.receiveSpeechText = "こんにちは，私の名前は翔太です。";
    this.isFirstTime = this.checkFirstTime(
      `lesson${this.lesson.number}:isFirstTime`
    );
    this.isStageFirstTime = this.checkFirstTime(
      `lesson${this.lesson.number}-stage${this.stage.number}`
    );

    // sessionStorageのrestore処理
    this.schemas = this.storage.restoreWorkspace(this.schemas);
  }

  restoreSubmittedWorkspace(schemas: DesignSchemaType): void {
    if (schemas) {
      this.schemas = schemas;
    }
  }

  checkFirstTime(url: string): boolean {
    try {
      const isFirstTime = window.localStorage.getItem(url);
      return isFirstTime !== "false";
    } catch {
      return true;
    }
  }

  setFirstTimeDone(url: string): void {
    this.isFirstTime = false;
    try {
      window.localStorage.setItem(url, "false");
    } catch {}
  }

  setClickElm(clickElm: ClickElement): void {
    this.clickElm = clickElm;
  }

  setStyle(className: string, style: ComponentStyleType): void {
    const prevStyle = toJS(this.schemas[className].style);
    const nextStyle = {
      ...prevStyle,
      ...style,
    };
    this.schemas[className].style = nextStyle;
    this.clickElm.style = nextStyle;

    this.storage.saveWorkspace(this.schemas);
  }

  setSendSpeechText(text: string): void {
    this.sendSpeechText = text;
  }

  setReceiveSpeechText(text: string): void {
    this.receiveSpeechText = text;
  }

  setStageConfig(config: StageConfig): void {
    this.stageConfig = config;
  }
}

decorate(Store, {
  clickElm: observable,
  schemas: observable,
  isFirstTime: observable,
  sendSpeechText: observable,
  receiveSpeechText: observable,
  setClickElm: action,
  setStageConfig: action,
  setStyle: action,
  setSendSpeechText: action,
  setReceiveSpeechText: action,
  setFirstTimeDone: action,
});

export default Store;
