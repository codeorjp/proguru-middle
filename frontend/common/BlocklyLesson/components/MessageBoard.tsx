import _ from "lodash";
import React from "react";
import { inject, observer } from "mobx-react";
import { toJS } from "mobx";
import flattenObject from "common/utils/flattenObject";
import SubmitArea from "common/BlocklyLesson/components/SubmitArea";
import styles from "common/BlocklyLesson/styles/MessageBoard.scss";
import ReadmoreButton from "common/BlocklyLesson/components/message_board/ReadmoreButton";
import Timeline from "common/BlocklyLesson/components/message_board/Timeline";
import ExecuteOverlay from "common/BlocklyLesson/components/message_board/ExecuteOverlay";

interface MessageBoardProps {
  store?: any;
  types: string[];
}

class MessageBoard extends React.Component<MessageBoardProps> {
  styleNames = [
    "SendSpeechBalloon",
    "SendProfileIcon",
    "ReceiveSpeechBalloon",
    "ReceiveProfileIcon",
    "ChatInput",
    "SubmitButton",
    "ChatBody",
  ];
  styles: {
    [key: string]: object;
  };

  constructor(props) {
    super(props);
    this.styles = {};
    this.getPastMessages = this.getPastMessages.bind(this);
  }

  getPastMessages() {
    this.props.store.getPastMessages();
  }

  render() {
    const { types } = this.props;
    const {
      messages,
      messageBoardStyle,
      isExecute,
      getPastMessages,
      existPastMessage,
      isShowExecuteOverlay,
    } = this.props.store;

    if (messageBoardStyle) {
      // レッスンで提出済みのスタイルにここで上書きされる
      this.styleNames.forEach((styleName) => {
        this.styles[styleName] = flattenObject(
          toJS(messageBoardStyle)[styleName].style
        );
      });
    }

    const sortedUniqueMessages = _.sortBy(_.uniqBy(messages, "messageId"), [
      "createdAt",
    ]);
    const {
      messageboard,
      messageboardWindow,
      messageboardBody,
      messageboardBodyHeader,
      messageboardName,
      messageboardContents,
      messageboardBodyFooter,
      messageboardMoreButton,
    } = styles;

    return (
      <div>
        <div className={messageboard}>
          <div className={messageboardWindow}>
            <div
              className={messageboardBody}
              style={messageBoardStyle ? this.styles.ChatBody : null}
            >
              <div className={messageboardBodyHeader}>
                <div className={messageboardName}>
                  <p>クラスのチャット</p>
                </div>
              </div>
              <div className={`${messageboardContents} messageBoardContents`}>
                <ReadmoreButton
                  isShow={!!getPastMessages}
                  handleClick={this.getPastMessages}
                  className={messageboardMoreButton}
                  isExecute={isExecute}
                  hasPastMessage={existPastMessage}
                />
                <Timeline
                  messages={sortedUniqueMessages}
                  userId={this.props.store.userId}
                  hasStudentsDesign={!!messageBoardStyle}
                  styles={this.styles}
                />
              </div>
              <div className={messageboardBodyFooter}>
                <SubmitArea types={types} />
              </div>
            </div>
          </div>
          <ExecuteOverlay
            isShow={isShowExecuteOverlay}
            isExecute={isExecute}
            styles={styles}
          />
        </div>
      </div>
    );
  }
}

export default inject("store")(observer(MessageBoard));
