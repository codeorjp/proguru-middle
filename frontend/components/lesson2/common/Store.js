import axios from "common/utils/axios";
import BlocklyJS from "blockly/javascript";
import moment from "moment";
import { decorate, observable } from "mobx";
import BlocklyLessonStore from "common/BlocklyLesson/BlocklyLessonStore";
import AnimationStore from "./AnimationStore";

class Store extends BlocklyLessonStore {
  constructor(userId, nickName, userIcon, lesson, stage, submittedWorkspace) {
    super(userId, submittedWorkspace);
    this.nickName = nickName;
    this.userIcon = userIcon;
    this.lesson = lesson;
    this.stage = stage;
    this.AnimationStore = AnimationStore;
    this.animate = this.animate.bind(this);
    this.receivedText = "";
    this.variables = {
      responseText: "",
      text: "",
    };
    this.servers = {
      serverA: "/api/say_hello",
      serverB: "cannot access serverB",
    };
    this.connectedServer = "";
    this.isFirstTime = this.checkFirstTime(
      `lesson${this.lesson.number}:isFirstTime`
    );
    this.isStageFirstTime = this.checkFirstTime(
      `lesson${this.lesson.number}-stage${this.stage.number}`
    );
  }

  init() {
    this.AnimationStore.reset();
    this.isExecute = false;
    this.receivedText = "";
    this.variables = {
      responseText: "",
      text: "",
    };
    this.connectedServer = "";
  }

  animate() {
    this.AnimationStore.animate();
  }

  addCommand(type, args, blockId) {
    this.AnimationStore.commands.push({
      type,
      args,
      blockId,
    });
  }

  skip(blockId) {
    this.addCommand("skip", null, blockId);
  }

  display(varName, blockId) {
    const msg = this.variables[varName];
    if (msg) {
      this.addCommand("display", { text: unescape(msg) }, blockId);
    }
  }

  assignText(text) {
    this.variables.text = text;
  }

  getMeridiem() {
    return moment().format("a");
  }

  connect(varName, callback, blockId, resolve) {
    if (varName && this.servers[varName]) {
      this.addCommand("connect", { before: "サーバと接続中..." }, blockId);

      if (varName === "serverA") {
        // serverAを選択してる場合のみ接続可能
        this.addCommand(
          "connectSuccess",
          { before: "接続成功しました！" },
          blockId
        );
        this.connectedServer = this.servers[varName];
        callback();
        resolve();
      } else {
        this.addCommand(
          "connectError",
          { before: "接続が失敗しました..." },
          blockId
        );
        resolve();
      }
    }
  }

  send(varName, blockId, resolve) {
    if (varName && this.variables[varName] && this.connectedServer) {
      this.addCommand(
        "request",
        { before: "メッセージを送信中...", after: "送信に成功しました！" },
        blockId
      );
      const formData = new FormData();
      formData.append("body", unescape(this.variables[varName]));
      formData.append("hour", new Date().getHours());
      axios.post(this.connectedServer, formData).then((res) => {
        this.receivedText = res.data.reply;
        resolve();
      });
    }
  }

  receiveResponse(callback, blockId) {
    if (this.receivedText !== "") {
      this.addCommand(
        "response",
        {
          before: "サーバからの応答を待ってます...",
          after: "応答を受け取りました！",
        },
        blockId
      );
      this.variables.responseText = this.receivedText;
      callback();
    }
  }
}

decorate(Store, {
  isFirstTime: observable,
});

export default Store;
