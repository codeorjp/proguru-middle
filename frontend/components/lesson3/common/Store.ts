import _ from "lodash";
import BlocklyJS from "blockly/javascript";
import moment from "moment";
import { action, decorate, observable } from "mobx";
import BlocklyLessonStore from "common/BlocklyLesson/BlocklyLessonStore";
import axios from "common/utils/axios";
import Endpoint from "common/utils/Endpoint";

class Store extends BlocklyLessonStore {
  nickName: string;
  studentNumber: number;
  userIcon: any; // blob?
  lesson: any;
  stage: any;
  submitCallback: Function | null;
  fetchServerID: number | null;
  isFirstTime: boolean;
  isStageFirstTime: boolean;
  messageBoardStyle: object;
  messages: object[];
  variables: any;
  last_id: number;

  constructor(
    userId,
    nickName,
    studentNumber,
    userIcon,
    lesson,
    stage,
    submittedWorkspace
  ) {
    super(userId, submittedWorkspace);
    this.nickName = nickName;
    this.studentNumber = studentNumber;
    this.userIcon = userIcon;
    this.lesson = lesson;
    this.stage = stage;
    this.fetchServerID = null;
    this.isFirstTime = this.checkFirstTime(
      `lesson${this.lesson.number}:isFirstTime`
    );
    this.isStageFirstTime = this.checkFirstTime(
      `lesson${this.lesson.number}-stage${this.stage.number}`
    );
    this.init();
    axios.get("/api/lessons/1/stages/8/workspaces").then((res) => {
      const { workspace } = res.data;
      this.messageBoardStyle = workspace ? JSON.parse(workspace.body) : null;
    });
  }

  init() {
    clearInterval(this.fetchServerID);
    this.isExecute = false;
    this.messages = [];
    this.submitCallback = async () => {};
    this.variables = {
      sendText: "",
      receivedText: "",
      newMessage: "",
      inputText: "",
      text: "",
    };
    this.last_id = null;
  }

  setChatInput(inputText: string) {
    this.variables.inputText = {
      messageId: _.uniqueId("message_not_sent_"),
      senderId: this.userId,
      senderType: "Student",
      userName: this.nickName,
      userNumber: this.studentNumber,
      icon: this.userIcon,
      content: inputText,
      createdAt: moment(),
      type: "Text",
    };
  }

  submit() {
    this.submitCallback().then((callback) => {
      if (callback) {
        callback(() => this.setChatInput(""));
      }
    });
  }

  setInputFileElement() {}

  // CommonToReceiveAndSend Function
  display(varName: string) {
    const message = this.variables[varName];
    if (message && message.content) {
      this.messages.push({
        ...message,
      });
    }
  }

  assignText(text: string) {
    this.variables.text = {
      messageId: _.uniqueId("message_not_sent_"),
      senderId: this.userId,
      senderType: "Student",
      userName: this.nickName,
      userNumber: this.studentNumber,
      icon: this.userIcon,
      content: unescape(text),
      createdAt: moment(),
      type: "Text",
    };
  }

  fetchMessage(callback: Function) {
    axios
      .get(Endpoint.message, {
        params: {
          kind: "lesson",
          last_id: this.last_id,
        },
      })
      .then((res) => {
        const { messages } = res.data;
        if (messages.length === 0) return;
        this.last_id = messages[messages.length - 1].id;

        _.map(messages, (message) => {
          const {
            id,
            sender_id,
            sender_type,
            nickname,
            number,
            icon_url,
            sent_at,
            body,
          } = message;
          this.variables.newMessage = {
            messageId: id,
            senderId: sender_id,
            senderType: sender_type,
            userName: nickname,
            userNumber: number,
            icon: icon_url,
            content: body,
            createdAt: moment(sent_at * 1000),
            type: "Text",
          };
          callback();
        });
      })
      .catch((error) => {
        console.log(error);
        clearInterval(this.fetchServerID);
      });
  }

  // Receive Function
  fetchServer(callback: Function) {
    this.fetchMessage(callback);
    this.fetchServerID = window.setInterval(() => {
      this.fetchMessage(callback);
    }, 5000);
  }

  isResponseIncludeText() {
    let isInclude = false;
    if (
      _.has(this.variables.newMessage, "content") &&
      this.variables.newMessage.content !== ""
    ) {
      isInclude = true;
    }
    return isInclude;
  }

  assignReceivedText(varName: string) {
    if (varName) {
      this.variables[varName] = this.variables.newMessage;
    }
  }

  // Send Function
  onSubmit(callback: Function) {
    this.submitCallback = callback;
  }

  send(varName: string, callback: Function, resolve: (value?: string) => void) {
    if (varName && this.variables[varName]) {
      if (this.variables[varName].content) {
        // テキストが空の場合は送信しない
        const params = new URLSearchParams();
        params.append("kind", "lesson");
        params.append("body", this.variables[varName].content);
        axios.post(Endpoint.message, params).then((res) => {
          this.variables[varName].messageId = res.data.id;
          this.variables[varName].createdAt = moment(res.data.created_at);
          callback();
          resolve();
        });
      }
    }
  }

  assignChatInput(varName: string) {
    if (varName) {
      this.variables[varName] = this.variables.inputText;
    }
  }
}

decorate(Store, {
  isFirstTime: observable,
  variables: observable,
  messages: observable,
  messageBoardStyle: observable,
  setChatInput: action,
  submit: action,
});

export default Store;
