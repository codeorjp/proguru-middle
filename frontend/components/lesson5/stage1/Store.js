import axios from "common/utils/axios";
import Lesson1Store from "components/lesson1/common/Store";

class Store extends Lesson1Store {
  constructor(userId, lesson, stage, submittedWorkspace) {
    super(userId, lesson, stage, submittedWorkspace);

    // 提出済みのデザインも SessionStorage に保存されているデザインもなければ 1-8 のデザインを取得する
    if (!this.submittedWorkspace && !this.storage.isExistsSavedWorkspace()) {
      axios.get("/api/lessons/1/stages/8/workspaces").then((res) => {
        const { workspace } = res.data;
        this.schemas = workspace ? JSON.parse(workspace.body) : this.schemas;
      });
    }
  }
}

export default Store;
