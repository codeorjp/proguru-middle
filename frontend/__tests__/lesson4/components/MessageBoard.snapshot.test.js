/* eslint-disable no-undef */
import React from "react";
import { Provider } from "mobx-react";
import renderer from "react-test-renderer";
import axios from "axios";
import Store from "components/lesson4/common/Store";
import MessageBoard from "common/BlocklyLesson/components/MessageBoard";
import BaseMessage from "../../mocks/mockMessage";

jest.mock("axios");
axios.get.mockResolvedValue({ data: { body: null } });

// see: https://stackoverflow.com/a/47781245
let dateNowSpy;
beforeAll(() => {
  // Lock Time
  dateNowSpy = jest.spyOn(Date, "now").mockImplementation(() => 1487076708000);
});

afterAll(() => {
  // Unlock Time
  dateNowSpy.mockRestore();
});

const userId = 0;
const otherUserId = 100;

const createStore = () =>
  new Store(
    userId, // userId
    "hoge", // nickname
    1, // studentNumber
    "usericon", // userIcon
    { number: 0 }, // lesson
    { number: 0 }, // stage
    null // submittedWorkspace
  );

test("snapshot", () => {
  const store = createStore();
  const component = renderer.create(
    <Provider store={store}>
      <MessageBoard types={["Image"]} />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("with messages", () => {
  const store = createStore();
  const SentTextMessage = new BaseMessage({
    messageId: 1,
    senderId: userId,
    type: "Text",
  });
  const SentImageMessage = new BaseMessage({
    messageId: 2,
    senderId: userId,
    type: "Image",
  });
  const ReceivedTextMessage = new BaseMessage({
    messageId: 3,
    senderId: otherUserId,
    type: "Text",
  });
  const ReceivedImageMessage = new BaseMessage({
    messageId: 4,
    senderId: otherUserId,
    type: "Image",
  });
  store.messages.push(
    SentTextMessage,
    SentImageMessage,
    ReceivedTextMessage,
    ReceivedImageMessage
  );
  const component = renderer.create(
    <Provider store={store}>
      <MessageBoard types={["Image"]} />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
