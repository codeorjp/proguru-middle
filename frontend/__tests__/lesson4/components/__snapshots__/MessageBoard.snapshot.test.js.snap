// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        style={null}
      >
        <div>
          <div>
            <p>
              クラスのチャット
            </p>
          </div>
        </div>
        <div
          className="undefined messageBoardContents"
        />
        <div>
          <div>
            <div
              style={
                Object {
                  "display": "block",
                }
              }
            >
              <button
                onClick={[Function]}
                type="button"
              >
                <input
                  accept="image/jpeg"
                  onChange={[Function]}
                  onClick={[Function]}
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                  type="file"
                />
                <i
                  className="bi bi-image undefined"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                />
              </button>
            </div>
            <div>
              <input
                disabled={true}
                onChange={[Function]}
                style={null}
                value=""
              />
            </div>
            <div>
              <button
                disabled={true}
                onClick={[Function]}
                style={null}
                type="button"
              >
                送信する
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <p>
        プログラムを組んだら
        <br />
        「実行する」ボタンを押してね
      </p>
    </div>
  </div>
</div>
`;

exports[`with messages 1`] = `
<div>
  <div>
    <div>
      <div
        style={null}
      >
        <div>
          <div>
            <p>
              クラスのチャット
            </p>
          </div>
        </div>
        <div
          className="undefined messageBoardContents"
        >
          <div>
            <div
              className="sendMessageArea"
            >
              <div>
                <p>
                  <span>
                    9:51 PM
                  </span>
                  <span>
                    出席番号：1
                  </span>
                  <span>
                    hogehoge
                  </span>
                </p>
                <div
                  style={null}
                >
                  <p>
                    test
                  </p>
                </div>
              </div>
              <div
                style={null}
              >
                <img
                  alt=""
                  src="iconurl"
                />
              </div>
            </div>
          </div>
          <div>
            <div
              className="sendMessageArea"
            >
              <div>
                <p>
                  <span>
                    9:51 PM
                  </span>
                  <span>
                    出席番号：1
                  </span>
                  <span>
                    hogehoge
                  </span>
                </p>
                <div
                  aria-describedby="popup-1"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": "0px",
                    }
                  }
                >
                  <img
                    alt=""
                    src="test"
                  />
                </div>
              </div>
              <div
                style={null}
              >
                <img
                  alt=""
                  src="iconurl"
                />
              </div>
            </div>
          </div>
          <div>
            <div
              className="sendMessageArea"
            >
              <div
                style={null}
              >
                <img
                  alt=""
                  src="iconurl"
                />
              </div>
              <div>
                <p>
                  <span>
                    出席番号：1
                  </span>
                  <span>
                    hogehoge
                  </span>
                  <span>
                    9:51 PM
                  </span>
                </p>
                <div
                  style={null}
                >
                  <p>
                    test
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              className="sendMessageArea"
            >
              <div
                style={null}
              >
                <img
                  alt=""
                  src="iconurl"
                />
              </div>
              <div>
                <p>
                  <span>
                    出席番号：1
                  </span>
                  <span>
                    hogehoge
                  </span>
                  <span>
                    9:51 PM
                  </span>
                </p>
                <div
                  aria-describedby="popup-2"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": "0px",
                    }
                  }
                >
                  <img
                    alt=""
                    src="test"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div>
            <div
              style={
                Object {
                  "display": "block",
                }
              }
            >
              <button
                onClick={[Function]}
                type="button"
              >
                <input
                  accept="image/jpeg"
                  onChange={[Function]}
                  onClick={[Function]}
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                  type="file"
                />
                <i
                  className="bi bi-image undefined"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                />
              </button>
            </div>
            <div>
              <input
                disabled={true}
                onChange={[Function]}
                style={null}
                value=""
              />
            </div>
            <div>
              <button
                disabled={true}
                onClick={[Function]}
                style={null}
                type="button"
              >
                送信する
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <p>
        プログラムを組んだら
        <br />
        「実行する」ボタンを押してね
      </p>
    </div>
  </div>
</div>
`;
